src/EditorFeatures/Core/SymbolSearch/SymbolSearchUpdateEngineFactory.cs

```diff
@@ -36,7 +38,34 @@ internal static partial class SymbolSearchUpdateEngineFactory
             }
 
             // Couldn't go out of proc.  Just do everything inside the current process.
-            return new SymbolSearchUpdateEngine(logService, progressService);
+            return CreateEngineInProcess(logService, progressService);
+        }
+
+        /// <summary>
+        /// This returns a No-op engine if called on non-Windows OS, becuase the backing storage dependes on Windows APIs.
+        /// </summary>
+        public static ISymbolSearchUpdateEngine CreateEngineInProcess(
+            ISymbolSearchLogService logService,
+            ISymbolSearchProgressService progressService)
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                ? new SymbolSearchUpdateEngine(logService, progressService)
+                : (ISymbolSearchUpdateEngine)new NoOpUpdateEngine();
+        }
+
+        private sealed class NoOpUpdateEngine : ISymbolSearchUpdateEngine
+        {
+            public Task<ImmutableArray<PackageWithAssemblyResult>> FindPackagesWithAssemblyAsync(string source, string assemblyName, CancellationToken cancellationToken)
+                => Task.FromResult(ImmutableArray<PackageWithAssemblyResult>.Empty);
+
+            public Task<ImmutableArray<PackageWithTypeResult>> FindPackagesWithTypeAsync(string source, string name, int arity, CancellationToken cancellationToken)
+                => Task.FromResult(ImmutableArray<PackageWithTypeResult>.Empty);
+
+            public Task<ImmutableArray<ReferenceAssemblyWithTypeResult>> FindReferenceAssembliesWithTypeAsync(string name, int arity, CancellationToken cancellationToken)
+                => Task.FromResult(ImmutableArray<ReferenceAssemblyWithTypeResult>.Empty);
+
+            public Task UpdateContinuouslyAsync(string sourceName, string localSettingsDirectory)
+                => Task.CompletedTask;
         }
```

Comment:
separate file please.

Reply:
Hmm, we have 3 private types defined in this file, `NoOpUpdateEngine`, `RemoteUpdateEngine` and `CallbackObject`, and they are only used within this file. I'd prefer to keep them here, unless separate files is required.